assets/js/admin/marketplace-suggestions.js

```diff
@@ -0,0 +1,411 @@
+					'product-edit-meta-tab-footer',
+					'product-edit-meta-tab-body'
+				]
+			};
+			var utmSource = _.findKey( sourceContextMap, function( sourceInfo ) {
+				return _.contains( sourceInfo, context );
+			} );
+			if ( utmSource ) {
+				utmParams.utm_source = utmSource;
+			}
+
+			return url + '?' + jQuery.param( utmParams );
+		}
+
+		// Render DOM element for suggestion linkout, optionally with button style.
+		function renderLinkout( context, slug, url, text, isButton ) {
+			var linkoutButton = document.createElement( 'a' );
+
+			var utmUrl = addUTMParameters( context, url );
+			linkoutButton.setAttribute( 'href', utmUrl );
+			linkoutButton.setAttribute( 'target', 'blank' );
+			linkoutButton.textContent = text;
+
+			linkoutButton.onclick = function( event ) {
+				window.wcTracks.recordEvent( 'marketplace_suggestion_clicked', {
+					suggestionSlug: slug
+				} );
+			}
+
+			if ( isButton ) {
+				linkoutButton.classList.add( 'button' );
+			}
+			else {
+				linkoutButton.classList.add( 'linkout' );
+				var linkoutIcon = document.createElement( 'span' );
+				linkoutIcon.classList.add( 'dashicons', 'dashicons-external' );
+				linkoutButton.appendChild( linkoutIcon );
+			}
+
+			return linkoutButton;
+		}
+
+		// Render DOM element for suggestion icon image.
+		function renderSuggestionIcon( iconUrl ) {
+			if ( ! iconUrl ) {
+				return null;
+			}
+
+			var image = document.createElement( 'img' );
+			image.src = iconUrl;
+			image.classList.add( 'marketplace-suggestion-icon' );
+
+			return image;
+		}
+
+		// Render DOM elements for suggestion content.
+		function renderSuggestionContent( title, copy ) {
+			var left = document.createElement( 'div' );
+
+			left.classList.add( 'marketplace-suggestion-container-content' );
+
+			if ( title ) {
+				var titleHeading = document.createElement( 'h4' );
+				titleHeading.textContent = title;
+				left.appendChild( titleHeading );
+			}
+
+			if ( copy ) {
+				var body = document.createElement( 'p' );
+				body.textContent = copy;
+				left.appendChild( body );
+			}
+
+			return left;
+		}
+
+		// Render DOM elements for suggestion call-to-action – button or link with dismiss 'x'.
+		function renderSuggestionCTA( context, slug, url, linkText, linkIsButton, allowDismiss ) {
+			var right = document.createElement( 'div' );
+
+			right.classList.add( 'marketplace-suggestion-container-cta' );
+			if ( url && linkText ) {
+				var linkoutElement = renderLinkout( context, slug, url, linkText, linkIsButton );
+				right.appendChild( linkoutElement );
+			}
+
+			if ( allowDismiss ) {
+				right.appendChild( renderDismissButton( slug ) )
+			}
+
+			return right;
+		}
+
+		// Render a "table banner" style suggestion.
+		// These are used in admin lists, e.g. products list.
+		function renderTableBanner( context, slug, iconUrl, title, copy, url, buttonText, allowDismiss ) {
+			if ( ! title || ! url ) {
+				return;
+			}
+
+			if ( ! buttonText ) {
+				buttonText = 'Go';
```

Comment:
This needs to be translated.

Reply:
Good spotting, this is potentially a dumb default anyway. I'll wrap it and change to the more common `Learn More`.

Reply:
This was one of those handy code review tips that revealed something less-than-ideal. I've now improved some related things:

- The default string is now `Learn More` – it's used for almost all content, so is a much better default
- The default applies for both button and link CTAs (was previously only button)
- The default CTA style is button, as the link is used much less often